crates/ruma-common/src/events/tag.rs

```diff
@@ -167,11 +170,15 @@ impl Serialize for TagName {
 }
 
 /// Information about a tag.
-#[derive(Clone, Debug, Default, Deserialize, Serialize)]
+#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
 #[cfg_attr(not(feature = "unstable-exhaustive-types"), non_exhaustive)]
 pub struct TagInfo {
     /// Value to use for lexicographically ordering rooms with this tag.
-    #[serde(skip_serializing_if = "Option::is_none")]
+    #[serde(
+        skip_serializing_if = "Option::is_none",
+        default,
+        deserialize_with = "deserialize_optional_f64_or_string"
+    )]
```

Comment:
Since this is not specified, it should be behind a `compat` feature:

```suggestion
    #[serde(skip_serializing_if = "Option::is_none")]
    #[cfg_attr(
        feature = "compat-tag-info",
        serde(default, deserialize_with = "deserialize_optional_f64_or_string")
    )]
```

Also might be nice to call it out in the field's documentation (`rg "If you activate"` for inspiration).